server/execute/switcher.go

```diff
@@ -0,0 +1,86 @@
+package execute
+
+import (
+	"encoding/json"
+	"log"
+	"net/http"
+	"strings"
+	"time"
+
+	"github.com/yanrishbe/http-server/entities"
+)
+
+func ManageLocation(location string, t time.Time, w http.ResponseWriter) {
```

Comment:
you can use a general function with the following signature:
func(location string) (time.Time, error)
You can reuse this function for both GET and POST requests. In case of the error, you can return a status code with an error.